census/src/main/java/io/grpc/census/CensusStatsModule.java

```diff
@@ -222,29 +263,34 @@ private void recordRealTimeMetric(TagContext ctx, MeasureLong measure, long valu
       inboundUncompressedSizeUpdater = tmpInboundUncompressedSizeUpdater;
     }
 
-    private final CensusStatsModule module;
-    final TagContext parentCtx;
-    private final TagContext startCtx;
-
     volatile long outboundMessageCount;
     volatile long inboundMessageCount;
     volatile long outboundWireSize;
     volatile long inboundWireSize;
     volatile long outboundUncompressedSize;
     volatile long inboundUncompressedSize;
-
-    ClientTracer(CensusStatsModule module, TagContext parentCtx, TagContext startCtx) {
-      this.module = checkNotNull(module, "module");
+    long roundtripNanos;
+    Code statusCode;
+
+    ClientTracer(
+        CallAttemptsTracerFactory attemptsState, CensusStatsModule module, TagContext parentCtx,
+        TagContext startCtx, StreamInfo info) {
+      this.attemptsState = attemptsState;
+      this.module = module;
       this.parentCtx = parentCtx;
-      this.startCtx = checkNotNull(startCtx, "startCtx");
+      this.startCtx = startCtx;
+      this.info = info;
+      this.stopwatch = module.stopwatchSupplier.get().start();
     }
 
     @Override
     public void streamCreated(Attributes transportAttrs, Metadata headers) {
       if (module.propagateTags) {
         headers.discardAll(module.statsHeader);
         if (!module.tagger.empty().equals(parentCtx)) {
-          headers.put(module.statsHeader, parentCtx);
+          if (!headers.containsKey(module.statsHeader)) {
```

Comment:
Why this change? Doesn't the `headers.discardAll()` make this check useless?

Reply:
reverted